confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -38,39 +40,66 @@
 
     Errors communicating to the server will result in a ClientError being raised.
 
-    @:param: url: url to schema registry
+    :param: str|dict url: url(deprecated) to schema registry or dictionary containing client configuration.
+    :param: str ca_location: File or directory path to CA certificate(s) for verifying the Schema Registry key.
+    :param: str cert_location: Path to client's public key used for authentication.
+    :param: str key_location: Path to client's private key used for authentication.
+
     """
 
     def __init__(self, url, max_schemas_per_subject=1000, ca_location=None, cert_location=None, key_location=None):
-        """Construct a client by passing in the base URL of the schema registry server"""
-
-        self.url = url.rstrip('/')
+        # In order to maintain compatibility the url(conf in future versions) param has been preserved for now.
+        conf = url
+        if not isinstance(url, dict):
+            conf = {
+                'url': url,
+                'ssl.ca.location': ca_location,
+                'ssl.certificate.location': cert_location,
+                'ssl.key.location': key_location
+            }
+            warnings.warn(
+                "CachedSchemaRegistry constructor is being deprecated. "
+                "Use CachedSchemaRegistryClient(dict: config) instead. "
+                "Existing params ca_location, cert_location and key_location will be replaced with their "
+                "librdkafka equivalents as keys in the conf dict: `ssl.ca.location`, `ssl.certificate.location` and "
+                "`ssl.key.location` respectively",
+                category=DeprecationWarning, stacklevel=2)
+
+        """Construct a Schema Registry client"""
+
+        # Ensure URL valid scheme is included; http[s]
+        if not conf.get('url', '').startswith('http'):
+            raise ValueError("Invalid URL provided for Schema Registry")
 
-        self.max_schemas_per_subject = max_schemas_per_subject
         # subj => { schema => id }
         self.subject_to_schema_ids = defaultdict(dict)
         # id => avro_schema
         self.id_to_schema = defaultdict(dict)
         # subj => { schema => version }
         self.subject_to_schema_versions = defaultdict(dict)
 
-        s = requests.Session()
-        if ca_location is not None:
-            s.verify = ca_location
-        if cert_location is not None or key_location is not None:
-            if cert_location is None or key_location is None:
-                raise ValueError(
-                    "Both schema.registry.ssl.certificate.location and schema.registry.ssl.key.location must be set")
-            s.cert = (cert_location, key_location)
+        s = Session()
+        s.verify = conf.pop('ssl.ca.location', None)
+        s.cert = self._configure_client_tls(conf)
+        s.auth = self._configure_basic_auth(conf)
 
+        self.url = conf.pop('url')
```

Comment:
shouldn't this check for error? (notexist)

Reply:
We already know it exists 

```
        # Ensure URL valid scheme is included; http[s]
        if not conf.get('url', '').startswith('http'):
            raise ValueError("Invalid URL provided for Schema Registry")
```